Ray.jl/test/task.jl

```diff
@@ -176,3 +176,17 @@ end
     pids2 = Ray.get.([submit_task(getpid, ()) for _ in 1:n_tasks])
     @test !isempty(intersect(pids, pids2))
 end
+
+@testset "task resource requests" begin
+    function gimme_resources()
+        resources = Ray.ray_jll.get_task_required_resources()
+        ks = Ray.ray_jll._keys(resources)
+        return Dict(String(k) => float(Ray.ray_jll._getindex(resources, k)) for k in ks)
+    end
+
+    default_resources = Ray.get(submit_task(gimme_resources, ()))
+    @test default_resources["CPU"] == 1.0
```

Comment:
what are the other keys that can be set?

Reply:
great question!  I tried using `memory` since that's special cased in the python remote function wrapper but it lead to tasks not being schedulable at all.  in theory you can use any key that you want, but your nodes have to declare that they have that resource for tasks to be scheduled there

Reply:
https://github.com/beacon-biosignals/ray/blob/4ceb62daaad05124713ff9d94ffbdad35ee19f86/python/ray/remote_function.py#L153-L164

Reply:
https://github.com/beacon-biosignals/ray/blob/4ceb62daaad05124713ff9d94ffbdad35ee19f86/src/ray/raylet/scheduling/scheduling_ids.h#L42-L45

Reply:
WAIT: https://github.com/beacon-biosignals/ray/blob/4ceb62daaad05124713ff9d94ffbdad35ee19f86/src/ray/protobuf/runtime_env_agent.proto#L28-L30

seems to imply that the units of memory request are GiB not bytes 🤦

Reply:
...but then there are things liek this in the tests:

https://github.com/beacon-biosignals/ray/blob/4ceb62daaad05124713ff9d94ffbdad35ee19f86/python/ray/tests/test_basic.py#L610

which suggests that the requests on teh user side are in bytes

Reply:
near as I can tell, the ptyhon remote function is doing exactly what we're doing: converting things to float adn dumping them into a dict in the task options: https://github.com/beacon-biosignals/ray/blob/4ceb62daaad05124713ff9d94ffbdad35ee19f86/python/ray/_raylet.pyx#L2899-L2909